### ISSUE TITLE
NiftiImageData3DTensor construtor from array not accessible

### ISSUE THREAD

**Issue by johannesmayer** [CCPPETMR/SIRF#303]
@rijobro 
The constructor 
``` 
template<class inputType>
NiftiImageData3DDisplacement(const inputType * const data, const VoxelisedGeometricalInfo3D &geom)
: NiftiImageData3DTensor<dataType>(data, geom) { this->check_dimensions(this->_3DDisp); }
```
asks for a `VoxelisedGeometricalInfo3D ` information, `check_dimensions` however, asks for a 5D image. So the test in the constructor always fails and one cannot pass a `VoxelisedGeometricalInfo5D`.

Should we do something, that the constructor `NiftiImageData3DTensor` automatically updates the geometry information to a 5D object with size 3 in the 5th direction? Is this straight-forward, or is there some trickery with for example with `_nifti_image->du`?

**Comment by johannesmayer** [comment]
I'm very happy with it!

**Comment by rijobro** [comment]
Ok, pushed to master. 

I also added `ctests` for `NiftiImageData3DTensor`, `NiftiImageData3DDisplacement` and `NiftiImageData3DDeformation` (as I'd already done for `NiftiImageData` and `NiftiImageData3D`). This highlighted another problem, which goes beyond the scope of this issue, but just so you're aware:

When we do an image comparison, we check if the image metadata match. If not, we resample them to be in the same space. But the NiftyReg resampler only works with 3D images, so this method won't work... 

TL;DR: Image comparison won't work for tensor images.

**Comment by rijobro** [comment]
Ignore bottom comment, managed to do resampling of tensor images too.

**Comment by johannesmayer** [comment]
Thank you very much, it compiles, have to see whether the performance will be correct though.
Will let you know!

**Comment by KrisThielemans** [comment]
probably ok!
